examples/key_value_store.rs

```diff
@@ -197,35 +197,32 @@ Options:
 
         /// Get data from the network.
         pub fn get(&mut self, what: String) {
-            let name = KeyValueStore::calculate_key_name(&what);
-            let data = self.example_client
-                .get(DataIdentifier::Structured(name, 10000));
-            match data {
-                Some(data) => {
-                    let sd = if let Data::Structured(sd) = data {
-                        sd
-                    } else {
-                        error!("KeyValueStore: Only storing structured data in this example");
-                        return;
-                    };
-                    if let Ok((key, value)) = deserialise::<(String, String)>(sd.get_data()) {
-                        println!("Got value {:?} on key {:?}", value, key);
-                    } else {
-                        error!("Failed to decode get response.");
-                        return;
-                    };
+            let name = Self::calculate_key_name(&what);
+            match self.example_client
+                      .get_mdata_value(name, TAG, what.as_bytes().to_vec()) {
+                Ok(value) => {
+                    let content = unwrap!(deserialise::<String>(&value.content));
+                    println!("Got value {:?} on key {:?}", content, what);
                 }
-                None => println!("Failed to get {:?}", what),
+                Err(error) => println!("Failed to get {:?} ({:?})", what, error),
             }
         }
 
         /// Put data onto the network.
-        pub fn put(&self, put_where: String, put_what: String) {
-            let name = KeyValueStore::calculate_key_name(&put_where);
-            let data = unwrap!(serialise(&(put_where, put_what)));
-            let sd = unwrap!(StructuredData::new(10000, name, 0, data, BTreeSet::new()));
-            if self.example_client.put(Data::Structured(sd)).is_err() {
-                error!("Failed to put data.");
+        pub fn put(&mut self, put_where: String, put_what: String) {
+            let name = Self::calculate_key_name(&put_where);
+
+            let key = put_where.into_bytes();
```

Comment:
I wonder whether it would be better to just use a constant (empty?) key.
Since this is an example, it should basically do what we would recommend for such a use-case; and I think using an empty key would make more sense than using `put_what` twice?